nri-redis/src/Redis/Internal.hs

```diff
@@ -198,7 +198,7 @@ query :: Stack.HasCallStack => Handler -> Query a -> Task Error a
 query handler query' =
   namespaceQuery (namespace handler ++ ":") query'
     |> Task.andThen (ensureMaxKeySize handler)
-    |> Task.andThen (Stack.withFrozenCallStack doQuery handler)
+    |> Task.andThen (Stack.withFrozenCallStack (doQuery handler))
```

Comment:
I don't totally understand `Stack.withFrozenCallStack` or execution order here (or really, deep skolemisation in general) 

But
- do we want to do this instead? Does it make a difference?
```suggestion
    |> Task.andThen (\query -> Stack.withFrozenCallStack ((doQuery handler) query))
```

Reply:
I don't think we want to do this. My partial understanding is that doing the above would prevent `withFrozenCallStack` from doing something useful (I've seen that happen in the past, though not  consistently).

The way I look at this (which might be wrong): Currently `withFrozenCallStack` is being passed a function, which makes sense because we want to modify that function's behavior to stop adding stack frames.

With the change you propose we'd be passing `withFrozenCallStack` a value that is the result of calling the function. i.e., by the time `withFrozenCallStack` is called, the function it wants to change the behavior of already ran, and we're too late.

Maybe our historical problems using `withFrozenCallStack` are related to use running strictly by default, and lazily evaluating functions would make it so `withFrozenCallStack` would work in the way you propose here.

Bottom line is I don't really understand how it works either. It's annoying and super easy to break accidentally in ways that makes our loggers start registering useless internal source locations.

I believe we have decent test coverage for this stuff though, so a "as long as the tests are happy" approach seems fair to me as well.